waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -26,15 +24,16 @@ import qualified Wasp.Util.Terminal as Term
 newtype ProjectName = ProjectName {_projectName :: String}
 
 createNewProject :: String -> Command ()
-createNewProject (all isLetter -> False) =
-  throwError $ CommandError "Project creation failed" "Please use only letters for a new project's name."
-createNewProject ((`elem` reservedNames) -> True) =
-  throwError $
-    CommandError
-      "Project creation failed"
-      ( "Please pick a project name not one of these reserved words:\n\t" ++ intercalate "\n\t" reservedNames
-      )
-createNewProject name = createNewProject' (ProjectName name)
+createNewProject name
+  | not (isValidWaspIdentifier name) =
+      throwError $ CommandError "Project creation failed" "The project's name must be a valid Wasp identifier."
```

Comment:
Should we provide instructions regarding what is a valid Wasp identifier? Probably, right?

Reply:
Yes, I thought about this but forgot to change it. Seems like we can't avoid duplicating the parsing rules after all :smile:

Reply:
Ha yes :D. We need to build a message based on logic in lexer ... -> kidding :D!